src/Definition.cpp

```diff
@@ -0,0 +1,210 @@
+#include <stdlib.h>
+
+#include "IR.h"
+#include "IRMutator.h"
+#include "Definition.h"
+#include "Var.h"
+
+namespace Halide {
+namespace Internal {
+
+using std::vector;
+using std::string;
+using std::map;
+
+struct DefinitionContents {
+    mutable RefCount ref_count;
+    bool is_init;
+    std::vector<Expr> values, args;
+    Schedule schedule;
+    ReductionDomain domain;
+    std::vector<Specialization> specializations;
+
+    DefinitionContents() : is_init(true) {}
+
+    void accept(IRVisitor *visitor) const {
+        for (Expr val : values) {
+            val.accept(visitor);
+        }
+        for (Expr arg : args) {
+            arg.accept(visitor);
+        }
+
+        schedule.accept(visitor);
+
+        if (domain.defined()) {
+            for (ReductionVariable rv : domain.domain()) {
+                rv.min.accept(visitor);
+                rv.extent.accept(visitor);
+            }
+            domain.predicate().accept(visitor);
+        }
+
+        for (const Specialization &s : specializations) {
+            if (s.condition.defined()) {
+                s.condition.accept(visitor);
+            }
+            s.definition.accept(visitor);
+        }
+    }
+
+    void mutate(IRMutator *mutator) {
+        for (size_t i = 0; i < values.size(); ++i) {
+            values[i] = mutator->mutate(values[i]);
+        }
+        for (size_t i = 0; i < args.size(); ++i) {
+            args[i] = mutator->mutate(args[i]);
+        }
+
+        schedule.mutate(mutator);
+
+        if (domain.defined()) {
+            domain.mutate(mutator);
```

Comment:
The 'accept' and 'mutate' both have the same number of calls. It's just that we didn't have ReductionDomain::accept. Just added a new 'accept' method inside ReductionDomain to make it more symmetric with the 'mutate'